### ISSUE TITLE
Bug: typing issue on `tseslint.config` with ESLint 9.29.0

### ISSUE THREAD

**Issue by threeal** [typescript-eslint/typescript-eslint#11308]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Repro Code

See: https://github.com/threeal/nodejs-starter/pull/839

### ESLint Config

```javascript
// eslint.config.ts

import eslint from "@eslint/js";
import { globalIgnores } from "eslint/config";
import tseslint from "typescript-eslint";

export default tseslint.config(
  globalIgnores(["dist"]),
  eslint.configs.recommended,
  tseslint.configs.strictTypeChecked,
  tseslint.configs.stylisticTypeChecked,
  {
    languageOptions: {
      parserOptions: {
        projectService: true,
        tsconfigRootDir: import.meta.dirname,
      },
    },
  },
);
```

### tsconfig

```jsonc
{
  "extends": "@tsconfig/node24",
  "compilerOptions": {
    "noEmit": true
  }
}
```

### Expected Result

Call to `tsc` does not cause typing error.

### Actual Result

Call to `tsc` result in the following:
```
eslint.config.ts:6:3 - error TS2345: Argument of type 'Config<RulesRecord>' is not assignable to parameter of type 'InfiniteDepthConfigWithExtends'.
  Type 'Config<RulesRecord>' is not assignable to type 'ConfigWithExtends'.
    Types of property 'languageOptions' are incompatible.
      Type 'import("/Users/threeal/Workspaces/nodejs/nodejs-starter/node_modules/.pnpm/eslint@9.29.0_jiti@2.4.2/node_modules/eslint/lib/types/index").Linter.LanguageOptions | undefined' is not assignable to type 'import("/Users/threeal/Workspaces/nodejs/nodejs-starter/node_modules/.pnpm/@typescript-eslint+utils@8.34.0_eslint@9.29.0_jiti@2.4.2__typescript@5.8.3/node_modules/@typescript-eslint/utils/dist/ts-eslint/Config").FlatConfig.LanguageOptions | undefined'.
        Type 'import("/Users/threeal/Workspaces/nodejs/nodejs-starter/node_modules/.pnpm/eslint@9.29.0_jiti@2.4.2/node_modules/eslint/lib/types/index").Linter.LanguageOptions' is not assignable to type 'import("/Users/threeal/Workspaces/nodejs/nodejs-starter/node_modules/.pnpm/@typescript-eslint+utils@8.34.0_eslint@9.29.0_jiti@2.4.2__typescript@5.8.3/node_modules/@typescript-eslint/utils/dist/ts-eslint/Config").FlatConfig.LanguageOptions'.
          Types of property 'ecmaVersion' are incompatible.
            Type 'EcmaVersion | undefined' is not assignable to type 'EcmaVersion'.
              Type '17' is not assignable to type 'EcmaVersion'.

6   globalIgnores(["dist"]),
    ~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in eslint.config.ts:6
```

Call to `eslint` does not have any issue, so purely typing issue. There's also no issue when using ESLint 9.28.0 or below.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `typescript-eslint`             | `8.34.0` |
| `TypeScript`                           | `5.8.3` |
| `ESLint`                               | `9.29.0` |
| `node`                                 | `24.2.0` |

**Comment by kirkwaiblinger** [comment]
The fix was released this morning, see https://github.com/typescript-eslint/typescript-eslint/releases/tag/v8.34.1. Please upgrade to the latest version
